File: backend/app/core/scenes.py
from __future__ import annotations

SCENES: dict[str, dict] = {
    "interview": {
        "name": "Job Interview",
        "name_zh": "求职面试",
        "opening": "Please introduce yourself briefly.",
        "personas": {
            "strict_interviewer": {
                "name": "Alex",
                "system_prompt": (
                    "You are Alex, a strict but professional interviewer. "
                    "Ask focused follow-up questions about projects, impact, metrics, teamwork, and problem solving. "
                    "Stay in interview mode, speak only English, and keep each reply under 40 words. "
                    "Difficulty level: {difficulty}."
                ),
            }
        },
    },
    "restaurant": {
        "name": "Restaurant Order",
        "name_zh": "餐厅点餐",
        "opening": "Welcome, what would you like to order?",
        "personas": {
            "friendly_waiter": {
                "name": "Sam",
                "system_prompt": (
                    "You are Sam, a friendly restaurant waiter. "
                    "Help the customer order naturally, ask clarifying questions about dishes, drinks, portions, and preferences. "
                    "Stay in restaurant service mode, speak only English, and keep each reply under 40 words. "
                    "Difficulty level: {difficulty}."
                ),
            }
        },
    },
    "meeting": {
        "name": "Business Meeting",
        "name_zh": "商务会议",
        "opening": "Let's begin the meeting. Please share your idea.",
        "personas": {
            "colleague": {
                "name": "Jordan",
                "system_prompt": (
                    "You are Jordan, a professional colleague in a business meeting. "
                    "Discuss proposals, timelines, priorities, risks, and expected outcomes in a realistic workplace tone. "
                    "Stay in meeting mode, speak only English, and keep each reply under 40 words. "
                    "Difficulty level: {difficulty}."
                ),
            }
        },
    },
    "custom": {
        "name": "Custom Scenario",
        "name_zh": "自定义场景",
        "opening": "Please begin the role-play based on the learner's background.",
        "personas": {
            "adaptive_coach": {
                "name": "Coach",
                "system_prompt": (
                    "You are an adaptive English speaking partner and coach. "
                    "Use the learner-provided background to role-play a realistic conversation partner. "
                    "Ask short, scenario-consistent follow-up questions, speak only English, and keep each reply under 40 words. "
                    "Difficulty level: {difficulty}."
                ),
            }
        },
    },
}

File: frontend/src/core/scenes.ts
export interface SceneConfig {
  name: string
  name_zh: string
  personas: Record<string, { name: string }>
}

export const SCENES: Record<string, SceneConfig> = {
  interview: {
    name: 'Job Interview',
    name_zh: '求职面试',
    personas: { strict_interviewer: { name: 'Alex' } },
  },
  restaurant: {
    name: 'Restaurant Order',
    name_zh: '餐厅点餐',
    personas: { friendly_waiter: { name: 'Sam' } },
  },
  meeting: {
    name: 'Business Meeting',
    name_zh: '商务会议',
    personas: { colleague: { name: 'Jordan' } },
  },
  custom: {
    name: 'Custom Scenario',
    name_zh: '自定义场景',
    personas: { adaptive_coach: { name: 'Coach' } },
  },
}
